core/src/main/scala/chisel3/Option.scala

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import chisel3.experimental.{BaseModule, SourceInfo}
+import chisel3.util.simpleClassName
+
+/** This object contains Chisel language definitions for describing configuration options and their accepted values.
+  */
+object option {
```

Comment:
I think this PR looks good, my only hangup is on the name of this object. It just feels very generic for this currently specific thing--and it isn't really related to what `Option` means in Scala more generally.

We will eventually want to add an Option type to Chisel (maybe called `Opt` to keep the pattern with `Vec` being Chisel's `Vector`), but that would probably live in just package `chisel3`.

Does anyone have any ideas? @nandor @azidar @seldridge @mwachs5?

Reply:
`option_group`?

Reply:
Maybe call it what it is: `chisel3.parameterization`?